lib/mongo/cluster.rb

```diff
@@ -359,8 +363,8 @@ def servers
     #
     # @since 2.1.0
     def disconnect!
```

Comment:
hmm I don't think you could get into a deadlock, disconnect! is not called by the garbage collector. I need to read the python ticket a few more times because I don't think I fully understand the scenario.

This did make me realize that I should called ``periodic_executor.stop!`` before calling #execute on it in the finalize method called by the garbage collector [here](https://github.com/mongodb/mongo-ruby-driver/pull/875/files#diff-e1d286ad6aafcf552750e39f66afcf57R203).